components/file_upload/file_upload.jsx

```diff
@@ -242,15 +250,22 @@ class FileUpload extends PureComponent {
     uploadFiles = (sortedFiles) => {
         const {currentChannelId, rootId} = this.props;
 
+        // Prohibts user from uploading .app files
+        const prohibitedFileTypesRegex = /^.*\.(app)/g;
```

Comment:
Although if we continue with checking for a file extension, this regex has some issues. Most notably, it can match file names that contain `.app` even if they don't end with it. There's also some potential performance issues by having it start with `^.*` as opposed to ending it with `$` along with some other trickier issues involving the `g` flag that I've run into with regexes like this in the past. I can go into more detail on those if you'd like, however I think we could just use `sortedFiles[i].name.endsWith('.app')` below instead of using a regex here.

Reply:
@hmhealey 
Hey Harrison! Thank you for the suggestion. I did what you said (remove regex and use `.endsWith()` however I couldn't do `endsWith('.app')` and had to do `endsWith('.app.zip')` because when I uploaded the `.app` file on chrome, It would get packaged as a `.zip` so it had the extra `.zip` added to it. I did notice a DRAMATIC improvement in the response times ( I was NOT aware that regex would cause such slowdowns 😓 ). Anyways, hope you can re-review and let me know what you think. I feel this is the best we can do with what we are given because the MIME type is `application/zip` so therefore I wasn't able to use the MIME type. Our drag and drop and ctrl+v already takes care of not allowing `.app` not to upload (since it's treated as a directory).